cpp/dolfin/fem/dofs_permutation.cpp

```diff
@@ -83,6 +91,89 @@ std::array<int, 4> calculate_tetrahedron_orders(int v1, int v2, int v3, int v4)
       "Two of a tetrahedron's vertices appear to be equal.");
 }
 //-----------------------------------------------------------------------------
+/// Calculates the number of times the rotation and reflection of a
+/// quadrilateral should be applied to a quadrilateral with the given global
+/// vertex numbers
+/// @param[in] v1 The global vertex number of the quadrilateral's first vertex
+/// @param[in] v2 The global vertex number of the quadrilateral's second vertex
+/// @param[in] v3 The global vertex number of the quadrilateral's third vertex
+/// @param[in] v4 The global vertex number of the quadrilateral's fourth vertex
+/// @return The rotation and reflection orders for the quadrilateral
+std::array<int, 2> calculate_quadrilateral_orders(int v1, int v2, int v3,
```

Comment:
This needs updating to 64-bit int (and others below).

Reply:
I've added templating to (I think) everywhere that a larger or smaller int is needed